src/components/Kanban/Kanban.tsx

```diff
@@ -0,0 +1,272 @@
+                      />
+                    ))}
+                    {provided.placeholder}
+                  </div>
+                  <div className="w-full">
+                    {!taskAddMode ? (
+                      <button
+                        className="flex w-full flex-row items-center gap-1 rounded-md px-1 text-left hover:bg-gray-200 dark:hover:bg-gray-600"
+                        onClick={() => setTaskAddMode(true)}
+                      >
+                        <BsPlus className="h-6 w-6" />
+                        <span className="align-middle">Add Task</span>
+                      </button>
+                    ) : (
+                      <form onSubmit={e => onFormSubmit(e)}>
+                        <div className="px-1">
+                          <input
+                            autoFocus
+                            value={taskInputValue}
+                            onChange={event => {
+                              setTaskInputValue(event.target.value);
+                            }}
+                            placeholder="Enter a task name..."
+                            className="mb-2 w-full overflow-hidden rounded-sm border border-gray-300 p-1 dark:border-gray-500 dark:bg-gray-700"
+                          />
+                          <div className="flex w-full flex-row gap-1">
+                            <button
+                              ref={addTaskButtonRef}
+                              className="rounded-md bg-blue-600 px-2 py-0.5 text-white hover:bg-blue-700"
+                            >
+                              Add Task
+                            </button>
+                            <button
+                              className="rounded-md px-2 py-0.5 hover:bg-gray-300 dark:hover:bg-gray-600"
+                              onClick={() => setTaskInputValue("")}
+                            >
+                              Cancel
+                            </button>
+                          </div>
+                        </div>
+                      </form>
+                    )}
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+        );
+      }}
+    </Droppable>
+  );
+};
+
+export const Kanban = ({}) => {
+  const { setIsKanbanToggled } = useToggleKanban();
+  const { board, setColumns } = useKanban();
+  const isDesktop = useMediaQuery("(min-width: 641px)");
+
+  const addTask = (columnIndex: number, taskName: string) => {
+    let columns = board.columns;
+    columns[columnIndex].tasks.push({ id: v4(), name: taskName });
+
+    setColumns(columns);
+  };
+
+  const deleteTask = (columnIndex: number, taskIndex: number) => {
+    let columns = board.columns;
+
+    if (!confirm(`Are you sure you want to delete the task "${columns[columnIndex].tasks[taskIndex].name}"?`))
+      return;
+
+    columns[columnIndex].tasks.splice(taskIndex, 1);
```

Comment:
I wonder if this can lead to some troublesome behaviour.
What happens at taskIndex == 0?

Reply:
When I was testing things seemed fine, I'll try for this specific case again and see.

Reply:
Tested this and it seems fine. Let me know if I'm missing something in my logic but taskIndex == 0 just means the first task in a column is deleted. If you're concerned about the "1" arg afterwards, that just means that we're explicitly removing only one single element from the array (which in this case would be at index 0, so the first one).